sdk/include/opentelemetry/sdk/trace/tracer_provider.h

```diff
@@ -18,12 +22,22 @@ class TracerProvider final : public opentelemetry::trace::TracerProvider
 {
 public:
   /**
-   * Initialize a new tracer provider.
+   * Initialize a new tracer provider with a default AlwaysOn sampler.
    * @param processor The span processor for this tracer provider. This must
    * not be a nullptr.
    */
   explicit TracerProvider(std::shared_ptr<SpanProcessor> processor) noexcept;
 
+  /**
+   * Initialize a new tracer provider with a specified sampler
+   * @param processor The span processor for this tracer provider. This must
+   * not be a nullptr.
+   * @param sampler The sampler for this tracer provider. This must
+   * not be a nullptr.
+   */
+  explicit TracerProvider(std::shared_ptr<SpanProcessor> processor,
+                          std::shared_ptr<Sampler> sampler) noexcept;
```

Comment:
How about default argument e.g.
std::shared_ptr<Sampler> sampler = make_shared<AlwaysOnEtc>
and then we don't need the other constructor above?